src/api/ZipDownload.ts

```
@@ -0,0 +1,102 @@
+        } catch (error: unknown) {
+            const errorMessage = error instanceof Error ? error.message : String(error);
+            throw new Error(`Failed to create ZIP download: ${errorMessage}`);
+        }
+    }
+
+    /**
+     * Download the ZIP file bytestream to the user's device using window.open
+     * @param bytestream - The ZIP file data as a bytestream
+     * @param filename - The name for the downloaded file
+     */
+    private downloadZipFile(url: any): void {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)